<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Working with Raw Data</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html"></a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="mini.html">Mini Modules</a>
</li>
<li>
  <a href="m1.html">Module One</a>
</li>
<li>
  <a href="m2.html">Module Two</a>
</li>
<li>
  <a href="m3.html">Module Three</a>
</li>
<li>
  <a href="res.html">Resources</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="http://github.com/popgenschwartz">
    <span class="fa fa-github fa-3x"></span>
     
  </a>
</li>
<li>
  <a href="https://www.schwartzlab-ecoevolutionarygenomics.org/">
    <span class="fa fa-school fa-3x"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Working with Raw Data</h1>

</div>


<div id="background" class="section level3">
<h3>Background</h3>
<p><font size="3"></p>
<p>Broad description of Module one and raw data. Broad description of Module One and raw data. Broad description of Module One and raw data. Broad description of Module One and raw data. Broad description of Module One and raw data. Broad description of Module One and raw data. Broad description of Module One and raw data. Broad description of Module One and raw data. Broad description of Module One and raw data.</p>
<p></font></p>
</div>
<div id="modules" class="section level1 tabset tabset-fade tabset-pills">
<h1><strong>Modules</strong></h1>
<div id="tutorial-1.1" class="section level2">
<h2>Tutorial 1.1</h2>
<p><font size="5"> <strong>Understanding your data</strong> </font></p>
<div id="blast_script" class="section level3">
<h3>BLAST_script</h3>
<p>BLAST stands for Basic Local Alignment Search Tool. There are quite a few different types of blast options: - BLASTN: Nucleotide - BLASTP: Protein - And less common BLASTX, TBLASTN, and TBLASTX</p>
<p>BLASTN, which is what is used in this spript, uses a nucleotide query and a nucleotide database. It matches oligos to a genome. A blast output generally provides raw scores, and Expect Value (E-value), and a list of alignments. However, there are many ways to limit the output, all of which can be found using <code>blastn --help</code>.</p>
<p>Output Options: &gt; Alignment view options:<br />
&gt; 0 = Pairwise<br />
&gt; 1 = Query-anchored showing identities<br />
&gt; 2 = Query-anchored no identities<br />
&gt; 3 = Flat query-anchored showing identities<br />
&gt; 4 = Flat query-anchored no identities<br />
&gt; 5 = BLAST XML<br />
&gt; 6 = Tabular<br />
&gt; 7 = Tabular with comment lines<br />
&gt; 8 = Seqalign (Text ASN.1)<br />
&gt; 9 = Seqalign (Binary ASN.1)<br />
&gt; 10 = Comma-separated values<br />
&gt; 11 = BLAST archive (ASN.1)<br />
&gt; 12 = Seqalign (JSON)<br />
&gt; 13 = Multiple-file BLAST JSON<br />
&gt; 14 = Multiple-file BLAST XML2<br />
&gt; 15 = Single-file BLAST JSON<br />
&gt; 16 = Single-file BLAST XML2<br />
&gt; 17 = Sequence Alignment/Map (SAM)</p>
<p>Option 7 was used to display the results in the tabular with comments format. Two limitations were used in our script to set a limit of one hit per query and the E-value to values less than 0.0001 (the smaller the E-value, the better):<br />
<code>-evalue</code> and <code>max_target_seqs</code></p>
<p>The general design of a blast from the command line is: <code>blastn -db &quot;DATABASE1&quot; DATABASE2&quot;... -query &quot;FILE.fasta&quot; -output &quot;OUT TYPE&quot; -[limit output: -evalue -Max_target_seqs or sorting of results]</code></p>
<p>Example of BLAST:<br />
<code>blastn -db &quot;ATmt.fasta ATcp.fasta ATchrV.fasta Arabidopsis_thaliana/CHR_I/NC_003070.gbk Arabidopsis_thaliana/CHR_II/NC_003071.gbk Arabidopsis_thaliana/CHR_III/NC_003074.gbk&quot; -query test.fasta -outfmt 7 -evalue 0.00001 -max_target-seqs 1| egrep -v '^#'| sed 's/[[:space:]]1_\/home.*NC_[0-9]*[[:space:]]/\tNT\t/'| awk '{print $1,$2}' | sort | uniq | awk '{print $2}' | sort | uniq -c | sort -n &gt; RawCounts.txt</code> In this code, the output of the number of hits from the mt, cp, and chrV fasta files were put into a file called RawCounts.txt</p>
<p>Finding “Dropped” sequences (No Hits): <code>NUM=$(blastn -db &quot;ATmt.fasta ATcp.fasta ATchrV.fasta Arabidopsis_thaliana/CHR_I/NC_003070.gbk Arabidopsis_thaliana/CHR_II/NC_003071.gbk Arabidopsis_thaliana/CHR_III/NC_003074.gbk&quot; -query test.fasta -outfmt 7 -evalue 0.00001 -max_target-seqs 1 | grep -c ' 0 hits ' ) &amp;&amp; echo No_hits $NUM &gt;&gt; RawCounts.txt</code> This counts the number of “No Hits” that occurred during the blast, and adds it to the RawCounts.txt file.</p>
<pre class="r"><code># !/bin/bash

module load samtools
module load bwa

#BWA

    #aligns
    #M= picard tools compatability
    #-v= verbose
    #T=Number of threads
    #order: reference, input, pipe to output
    #sort
    #Sbu sam to bam file

bwa mem -M  -v 2 -t 6 -R &quot;@RG\tID:Rep9Gen623_1\tSM:Rep9Gen6\tPL:illumina\tPU:_1\tLB:SRR1693723_1&quot; sacCer3 /home/aubcls35/Group_Project/Part_1/SRR1693723.C1630ACXX6.fastq.gz | samtools view -Sb | samtools sort &gt; SRR1693723_1.sorted.bam
bwa mem -M  -v 2 -t 6 -R &quot;@RG\tID:Rep9Gen623_2\tSM:Rep9Gen6\tPL:illumina\tPU:_2\tLB:SRR1693723_2&quot; sacCer3 /home/aubcls35/Group_Project/Part_1/SRR1693723.C1AC9ACXX5.fastq.gz | samtools view -Sb | samtools sort &gt; SRR1693723_2.sorted.bam
bwa mem -M  -v 2 -t 6 -R &quot;@RG\tID:Rep9Gen623_3\tSM:Rep9Gen6\tPL:illumina\tPU:_3\tLB:SRR1693723_3&quot; sacCer3 /home/aubcls35/Group_Project/Part_1/SRR1693723.C1AC9ACXX6.fastq.gz | samtools view -Sb | samtools sort &gt; SRR1693723_3.sorted.bam
bwa mem -M  -v 2 -t 6 -R &quot;@RG\tID:Rep9Gen624_1\tSM:Rep9Gen6\tPL:illumina\tPU:_1\tLB:SRR1693724_1&quot; sacCer3 /home/aubcls35/Group_Project/Part_1/SRR1693724.C1630ACXX6.fastq.gz | samtools view -Sb | samtools sort &gt; SRR1693724_1.sorted.bam
bwa mem -M  -v 2 -t 6 -R &quot;@RG\tID:Rep9Gen624_2\tSM:Rep9Gen6\tPL:illumina\tPU:_2\tLB:SRR1693724_2&quot; sacCer3 /home/aubcls35/Group_Project/Part_1/SRR1693724.C1AC9ACXX5.fastq.gz | samtools view -Sb | samtools sort &gt; SRR1693724_2.sorted.bam
bwa mem -M  -v 2 -t 6 -R &quot;@RG\tID:Rep9Gen624_3\tSM:Rep9Gen6\tPL:illumina\tPU:_3\tLB:SRR1693724_3&quot; sacCer3 /home/aubcls35/Group_Project/Part_1/SRR1693724.C1AC9ACXX6.fastq.gz | samtools view -Sb | samtools sort &gt; SRR1693724_3.sorted.bam
bwa mem -M  -v 2 -t 6 -R &quot;@RG\tID:Rep9Gen628_1\tSM:Rep9Gen6\tPL:illumina\tPU:_1\tLB:SRR1693728_1&quot; sacCer3 /home/aubcls35/Group_Project/Part_1/SRR1693728.C1AC9ACXX2.fastq.gz | samtools view -Sb | samtools sort &gt; SRR1693728_1.sorted.bam
bwa mem -M  -v 2 -t 6 -R &quot;@RG\tID:Rep9Gen628_2\tSM:Rep9Gen6\tPL:illumina\tPU:_2\tLB:SRR1693728_2&quot; sacCer3 /home/aubcls35/Group_Project/Part_1/SRR1693728.C1AC9ACXX3.fastq.gz | samtools view -Sb | samtools sort &gt; SRR1693728_2.sorted.bam
bwa mem -M  -v 2 -t 6 -R &quot;@RG\tID:Rep9Gen628_3\tSM:Rep9Gen6\tPL:illumina\tPU:_3\tLB:SRR1693728_3&quot; sacCer3 /home/aubcls35/Group_Project/Part_1/SRR1693728.D19EFACXX1.fastq.gz | samtools view -Sb | samtools sort &gt; SRR1693728_3.sorted.bam

    #index the samples above
samtools index -b SRR1693723_1.sorted.bam
samtools index -b SRR1693723_2.sorted.bam
samtools index -b SRR1693723_3.sorted.bam
samtools index -b SRR1693724_1.sorted.bam
samtools index -b SRR1693724_2.sorted.bam
samtools index -b SRR1693724_3.sorted.bam
samtools index -b SRR1693728_1.sorted.bam
samtools index -b SRR1693728_2.sorted.bam
samtools index -b SRR1693728_3.sorted.bam

#Merging
    #Finally merge these together and sort and index the merged bam file to assess the quality using samtools
    #Have the indexed files in the same directory as the bam files

samtools merge -r SRR1693723.merged.bam SRR1693723_1.sorted.bam SRR1693723_2.sorted.bam SRR1693723_3.sorted.bam
samtools merge -r SRR1693724.merged.bam SRR1693724_1.sorted.bam SRR1693724_2.sorted.bam SRR1693724_3.sorted.bam
samtools merge -r SRR1693728.merged.bam SRR1693728_1.sorted.bam SRR1693728_2.sorted.bam SRR1693728_3.sorted.bam

samtools sort SRR1693723.merged.bam &gt; SRR1693723.merged.final.bam
samtools sort SRR1693724.merged.bam &gt; ~/Group_Project/Part_2/Final_output/SRR1693724.merged.final.bam
samtools sort SRR1693728.merged.bam &gt; ~/Group_Project/Part_2/Final_output/SRR1693728.merged.final.bam

samtools index -b SRR1693723.merged.final.bam
samtools index -b ~/Group_Project/Part_2/Final_output/SRR1693724.merged.final.bam
samtools index -b ~/Group_Project/Part_2/Final_output/SRR1693728.merged.final.bam</code></pre>
</div>
</div>
<div id="tutorial-1.2" class="section level2">
<h2>Tutorial 1.2</h2>
</div>
<div id="tutorial-1.3" class="section level2">
<h2>Tutorial 1.3</h2>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
